install/kubernetes/cilium/templates/cilium-operator-deployment.yaml

```diff
@@ -78,6 +78,13 @@ spec:
       - args:
         - --config-dir=/tmp/cilium/config-map
         - --debug=$(CILIUM_DEBUG)
+{{- range $key, $value := .Values.operator.extraArgs }}
+{{- if $value }}
+        - --{{ $key }}={{ $value }}
+{{- else }}
+        - --{{ $key }}
+{{- end }}
+{{- end }}
```

Comment:
Oooh, this is a clever approach! I suppose `extraArgs` are then setup with something like this?

```yaml
operator:
    extraArgs:
        option1: value1
        option2: value2
        flag1:
        flag2:
```

It looks neat, but I would tend to think using a simple array would be more robust:

```yaml
operator:
    extraArgs:
        - --option1=value1
        - --option2=value2
        - --flag1
        - -f2
```

Which would be inserted with something like above for annotations:

```go
{{- with .Values.operator.extraArgs }}
        {{- toYaml . | nindent 8 }}
{{- end }}
```

In particular, this would allow usage of any arbitrary command line arguments (e.g. in short syntax `-v`), and not only those formatted as expected by `--key=value` or `--key`. I think it is more flexible :)

What do you and other reviewers think?

Reply:
All other templates use the code above...

Reply:
Sorry, I am new :D You are right: I missed that there was one instance of `extraArgs` already set up this way in `cilium-agent-daemonset.yaml`. That said, it seems to be the only one. The other `extraArgs` values are seemingly never used either, so perhaps we should take advantage of this PR to standardize and retrofit to all `extraArgs` instances, whatever decision is made re: object vs. array 🤔 

I will wait for other reviewers to chime in on this point.

Reply:
Agreed with @Skymirrh here.

Reply:
what is there? use `toYaml`?

Reply:
I'm waiting @seanmwinn sentiment on the topic.

Also, I would like to semi-retract something:

> perhaps we should take advantage of this PR to standardize and retrofit to all `extraArgs` instances

This was poorly conveyed: we should not expect you to do that refactoring. Of course if you feel like refactoring all instance, you're more than welcome, but if you don't, then that's also OK. All contributions are welcome :)

Reply:
I personally think that simple arrays is indeed a better options, but given that the proposed change is based on code pattern that existed elsewhere, I think it's reasonable to merge this PR as it stands :)

Reply:
I'm fine with whatever is going to be standardized across the entire chart. There are arguments for/against using either pattern. Feedback here indicates users are more in favor of using an array, which is fine.

I would just add that extraArgs is traditionally a development-cycle flag. It is useful for injecting flags for a binary at runtime. If we expect users to be able to tune or configure a setting regularly, I would suggest exposing a Helm value to make it tunable instead of relying on a long list of extraArgs.